marshmallow/schema.py

```diff
@@ -525,6 +525,16 @@ def dump(self, obj, many=None, update_fields=True, **kwargs):
             error_handler = self.handle_error or self.__error_handler__
             if callable(error_handler):
                 error_handler(errors, obj)
+            exc = ValidationError(
+                errors,
+                field_names=self._marshal.error_field_names,
+                fields=self._marshal.error_fields,
+                data=result,
+                **self._marshal.error_kwargs
+            )
+            self.handle_error(exc, result)
```

Comment:
Not correct. Yes, you will have errors. With `strict=True` you have to throw them as an exception (regardless of exception type), with `strict=False` you should not throw exception and just return errors. The thing is that if you handle errors with `self.handle_error` there is no guarantee that it won't throw exception. According to code, result of `self.handle_error` is always ignored. Although you might modify original error dictionary, this is not a common case. The common case is to throw an application-specific exception. Which violates strictness in some cases, because `self.strict` is not taken into account.

Anyways, now it makes even less sense: before it used either `self.handle_error` OR `self.__error_handler__`, now you call `self.__error_handler__`, and then `self.handle_error` without even checking if `self.handle_error` is callable.